azure/services/vmextensions/vmextensions.go

```diff
@@ -55,13 +58,29 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	defer span.End()
 
 	for _, extensionSpec := range s.Scope.VMExtensionSpecs() {
-		if _, err := s.client.Get(ctx, s.Scope.ResourceGroup(), extensionSpec.VMName, extensionSpec.Name); err == nil {
-			// check for the extension and don't update if already exists
-			// TODO: set conditions based on extension status
+		if existing, err := s.client.Get(ctx, s.Scope.ResourceGroup(), extensionSpec.VMName, extensionSpec.Name); err == nil {
+			// check the extension status and set the associated conditions.
+			switch compute.ProvisioningState(to.String(existing.ProvisioningState)) {
```

Comment:
Can we follow a similar pattern like we do when updating VM state and move this logic to scope?, ie.
```
scope.setVMExtState()
scope.updateStatus()
```

Maybe we can do it as part of future refactoring.